db/system_keyspace.cc

```diff
@@ -1534,6 +1564,71 @@ future<replay_positions> system_keyspace::get_truncated_positions(table_id cf_id
     co_return result;
 }
 
+future<> system_keyspace::drop_all_range_cleanup_records() {
+    sstring req = format("SELECT shard from system.{}", COMMITLOG_CLEANUPS);
+    auto rs = co_await execute_cql(req);
+
+    bool any = false;
+    co_await coroutine::parallel_for_each(rs->begin(), rs->end(), [&] (const cql3::untyped_result_set_row& row) -> future<> {
```

Comment:
It's worth adding a comment why deleting only the existing partitions is preferred over issuing a DELETE for each shard, unconditionally.

Reply:
No particular reason.

But note that we have to issue DELETEs for each shard of the past boot cycle, which could have a different number of shards than the current cycle. So we would have to pass the shard count of the past boot cycle to this function via an argument.

Reply:
> No particular reason.
> 
> But note that we have to issue DELETEs for each shard of the past boot cycle, which could have a different number of shards than the current cycle. So we would have to pass the shard count of the past boot cycle to this function via an argument.

This is a compelling reason

Reply:
I added a code comment to clarify this.